utils/pennsylvania.py

```diff
@@ -0,0 +1,446 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+
+import pandas as pd
+import numpy as np
+from utils import pennsylvania_helper_functions as helper
+from utils import clean
+import uuid
+# sys.path.append("/home/alankagiri/2023-fall-clinic-climate-cabinet")
+from utils import constants as const
+
+
+class PennsylvaniaCleaner(clean.StateCleaner):
+
+    curr_python_file = Path(__file__).resolve()
+    # resolve to the path of the current python file
+    # repo_root will be the absolute path to the root of the repository,
+    # no matter where the repository is.
+    # For pathlib Path objects, `.parent` gets the parent path_splitectory and `/`
+    # can be used like `/` in unix style paths.
+    repo_root = curr_python_file.parent.parent
+    filepaths_list = pd.read_csv(repo_root / "data" / "results.csv")
+
+    # helper functions: 
+    def assign_col_names(self,filepath: str, year: int) -> list:
+        """Assigns the right column names to the right datasets.
+
+        Args:
+            filepath: the path in which the data is stored/located.
+
+            year: to make parsing through the data more manageable, the year from
+            which the data originates is also taken.
+
+        Returns:
+            a list of the appropriate column names for the dataset
+        """
+        dir = filepath.split("/")
+        file_type = dir[len(dir) - 1]
+
+        if "contrib" in file_type:
```

Comment:
if contrib in file_type s and year > 2022 --> makes the code a little more clean if you group conditionals

Reply:
you can apply this logic to everything below

Reply:
But if I group the conditionals won't the same redundancy be there? Bc it would then be:
`if ('contrib)' in file_type) and (year<2022):
....
elif ('contrib' in file_type):
....`
either way I'd be checking for the word 'contrib' twice when the existing code does that only once...